MF real account creation function using API

bertrand-deriv opened the pull request:
### MF real account creation function using API

- Refactored the CR and VR real account creation by moving hardcoded data to fixtures
- Parameterized most of the client details e.g first and last name, dob, phone, address, tax residence, etc
- Created the MF real account creation function using API

<img width="511" alt="Screenshot 2024-07-02 at 12 49 30" src="https://github.com/deriv-com/e2e-deriv-app/assets/133766175/60222552-e7f8-42d1-95f3-b539b3831648">
<img width="540" alt="Screenshot 2024-07-02 at 12 58 45" src="https://github.com/deriv-com/e2e-deriv-app/assets/133766175/7d9e2708-f14c-4c1c-8a7c-734ff7d939cf">
<img width="916" alt="Screenshot 2024-07-02 at 12 59 04" src="https://github.com/deriv-com/e2e-deriv-app/assets/133766175/4accf026-669a-4f28-86ad-3a39cbf888cc">
<img width="1728" alt="Screenshot 2024-07-02 at 13 31 00" src="https://github.com/deriv-com/e2e-deriv-app/assets/133766175/4baaf400-cff9-4c08-94ad-da6966fa608f">

meeyun-deriv reviewed (commented):
can you try running some tests to make sure nothing breaks? you can attach the local run result in the PR description.

Code wise LGTM

bilalmajid-deriv commented on cypress/support/commands/common.js:
```diff
@@ -474,70 +474,97 @@ Cypress.Commands.add('c_walletLogout', () => {
-          //Reset oAuthUrl otherwise it will use the previous URL
-          Cypress.env('oAuthUrl', '<empty>')
-        })
+Cypress.Commands.add('c_createRealAccount', (clientData) => {
```
can you help to update all the test that are using this command to use the parameters accordingly because if this is merged then all those tests will start failing

bilalmajid-deriv replied:
can you also update it to be createCRAccount similar to createMFAccount
or may be merge both commands as createRealAccount () and pass MF or CR 

which ever is better

bertrand-deriv replied:
@bilalmajid-deriv yeah thanks for the review, Let me work on them.
Regarding the second comment, I will go with the option to rename to `createCRAccount`

bilalmajid-deriv commented on cypress/support/commands/common.js:
```diff
@@ -474,70 +474,97 @@ Cypress.Commands.add('c_walletLogout', () => {
+  Usage cy.c_createDemoAccount({country_code: 'ke'}) or you may not pass in anything to go with default.
+  Works either CR or MF countries.
+*/
+Cypress.Commands.add('c_createDemoAccount', (clientData) => {
```
can you also update tests to pass parameters expected for this command as well

bilalmajid-deriv reviewed (commented):
Left some comments

abuzar-deriv reviewed (approved):
LGTM apart from the existing comments.

bertrand-deriv commented:
@bilalmajid-deriv @meeyun-deriv 
I have implemented the changes to the relevant test cases.
I also run different test cases to make sure that tests are not failing due to these changes
Added the screenshot in description

bilalmajid-deriv commented on cypress/e2e/smoke/TradersHub/addSiblingsAccounts.cy.js:
```diff
@@ -25,7 +25,7 @@ describe('QATEST-5797, QATEST-5820: Add siblings accounts', () => {
   let taxIDNum = Cypress.env('taxIDNum').CO
   beforeEach(() => {
-    cy.c_createDemoAccount(countryCode, currencyCode)
+    cy.c_createDemoAccount({ country_code: countryCode })
```
@bertrand-deriv is the currency removed because it is a default value?

bertrand-deriv replied:
no, during demo account creation, we don't pass in currency..... This was incorrect usage

meeyun-deriv reviewed (approved):
LGTM
